packages/typespec-client-generator-core/src/types.ts

```diff
@@ -1363,46 +1280,83 @@ export function getSdkModelPropertyType(
-      if (httpBody) {
-        const httpBodyType = isHttpBodySpread(httpBody)
-          ? getHttpBodySpreadModel(httpBody.type as Model)
-          : httpBody.type;
-        if (type.model === httpBodyType) {
-          // only try to add multipartOptions for property of body
-          diagnostics.pipe(updateMultiPartInfo(context, type, property, operation));
-          property.multipartOptions = property.serializationOptions.multipart; // eslint-disable-line @typescript-eslint/no-deprecated
-        }
-      }
-    }
   }
   return diagnostics.wrap(property);
 }
 
 function addPropertiesToModelType(
   context: TCGCContext,
   type: Model,
-  sdkType: SdkType,
+  sdkType: SdkModelType,
   operation?: Operation,
 ): [void, readonly Diagnostic[]] {
   const diagnostics = createDiagnosticCollector();
   for (const property of type.properties.values()) {
     if (
       isStatusCode(context.program, property) ||
       isNeverOrVoidType(property.type) ||
-      hasNoneVisibility(context, property) ||
-      sdkType.kind !== "model"
+      hasNoneVisibility(context, property)
     ) {
       continue;
     }
     const clientProperty = diagnostics.pipe(getSdkModelPropertyType(context, property, operation));
-    if (sdkType.properties) {
-      sdkType.properties.push(clientProperty);
+    sdkType.properties.push(clientProperty);
+  }
+  return diagnostics.wrap(undefined);
+}
+
+function addMultipartPropertiesToModelType(
+  context: TCGCContext,
+  sdkType: SdkModelType,
+  operation: Operation,
+): [void, readonly Diagnostic[]] {
+  const diagnostics = createDiagnosticCollector();
+  const body = getHttpOperationWithCache(context, operation).parameters.body;
+  if (!body || body.bodyKind !== "multipart") {
+    return diagnostics.wrap(undefined);
+  }
+  for (const part of body.parts) {
+    const clientProperty = diagnostics.pipe(
+      getSdkModelPropertyType(context, part.property!, operation),
+    );
+
+    // set the type of the client property based on the part body type
+    const bodyType = getHttpBodyType(part.body);
+    if (clientProperty.type.kind === "array") {
+      clientProperty.type.valueType = diagnostics.pipe(
+        getClientTypeWithDiagnostics(context, bodyType, operation),
+      );
     } else {
-      sdkType.properties = [clientProperty];
+      clientProperty.type = diagnostics.pipe(
+        getClientTypeWithDiagnostics(context, bodyType, operation),
+      );
     }
+
+    clientProperty.serializationOptions.multipart = {
+      isFilePart: isFilePart(context, clientProperty.type),
+      isMulti: part.multi,
+      filename: part.filename
+        ? diagnostics.pipe(getSdkModelPropertyType(context, part.filename, operation))
+        : undefined,
+      contentType: part.body.contentTypeProperty
+        ? diagnostics.pipe(
+            getSdkModelPropertyType(context, part.body.contentTypeProperty, operation),
+          )
+        : undefined,
+      defaultContentTypes: part.body.contentTypes,
```

Comment:
Does a
```
  stringWithContentType: HttpPart<{
    @body body: string;
    @header contentType: "text/html";
  }>;
```

affect the `contentType` and `defaultContentTypes` here?

Reply:
Yes. The `contentType` is `contentType` property and `defaultContentTypes` is `["text/html"]`. But the headers will not contain the `contentType` header.